saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -639,6 +657,17 @@ public Builder singleSignOnServiceBinding(Saml2MessageBinding singleSignOnServic
 				return this;
 			}
 
+			/**
+			 * Set the signature algorithms to use when signing, in preference order.
+			 * @param signAlgorithmUris list of signature algorithm URI
+			 * @return this {@code Builder}
+			 * @since 5.5
+			 */
+			public Builder signAlgorithmUris(List<String> signAlgorithmUris) {
```

Comment:
For consistency with the rest of the `Builder`, let's have this take a `Consumer<Collection<String>>` instead of a `List<String>`. You can see the `signingX509Credentials` method for an example.

The reason that using a `Consumer` is nice is it makes it simpler to return an instance of `Builder` in an API where an application can add algorithms to the existing list.